fastly/base_fastly_service_v1.go

```diff
@@ -450,7 +450,12 @@ func resourceServiceRead(d *schema.ResourceData, meta interface{}, serviceDef Se
 	d.Set("version_comment", s.Version.Comment)
 	d.Set("active_version", s.ActiveVersion.Number)
 
-	if s.ActiveVersion.Number == 0 && isImport {
+	// If we are importing or `activate` is set to false, temporarily set the
+	// service.ActiveVersion number to the latest version supplied via the get
+	// service version details call. This is to ensure we still read all of the
+	// state below.
+	shouldActivate := d.Get("activate").(bool)
+	if s.ActiveVersion.Number == 0 && isImport || !shouldActivate {
```

Comment:
I'm on  the fence if this this is the cleanest way of solving this, or instead changing the gate on read conditional logic on line 165. Thoughts?

Reply:
Let's talk through the alternative implementation first before we go ahead and approve/merge this changeset.